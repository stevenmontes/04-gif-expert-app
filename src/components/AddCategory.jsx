import { useState } from "react"

export const AddCategory = ({ onAddCategory }) => {
    const [inputValue, setInputValue] = useState('');

    const onInputChange = (event) => {
        setInputValue(event.target.value);
    }

    const onSubmit = (e) => {
        e.preventDefault();

        if (inputValue.trim().length <= 1) return;

        onAddCategory(categories => [inputValue, ...categories]);
        setInputValue('');
    }

    return (
        <form onSubmit={(event) => onSubmit(event)}>
            <input type="text" placeholder="Buscar gifs" value={inputValue}
                onChange={onInputChange} />
        </form>
    )
}
