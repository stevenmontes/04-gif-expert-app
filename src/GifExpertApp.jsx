import { useState } from 'react';
import { AddCategory, GifGrid } from './components'

export const GifExpertApp = () => {

    const [categories, setCategories] = useState(['One Punch', 'Dragon Ball']);

    const onAddCategory = (newCategory) => {
        if (categories.includes(newCategory)) return;

        setCategories([newCategory, ...categories]);
    }

    return (
        <>
            <h1>Gif</h1>
            <button onClick={onAddCategory}>Agregar</button>
            <AddCategory onNewCategory={event => onAddCategory(event)}></AddCategory>
            {
                categories.map(category =>
                    (<GifGrid key={category} category={category}></GifGrid>)
                )
            }
        </>
    )
}
