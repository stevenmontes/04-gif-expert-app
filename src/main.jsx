import React from 'react'
import ReactDOM from 'react-dom/client'
import { GifExpertApp } from '../src/GifExpertApp'

ReactDOM.createRoot(document.getElementById('root')).render(
  <React.StrictMode>
    <GifExpertApp></GifExpertApp>
  </React.StrictMode>,
)
