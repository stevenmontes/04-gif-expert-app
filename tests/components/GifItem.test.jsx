import { render } from "@testing-library/react";
import { GifItem } from "../../src/components";

describe('Tests of GifItem', () => {
    test('should match with the snapshot', () => {
        const image = {
            title: 'title',
            url: 'url'
        }
        const { container } = render(<GifItem {...image}></GifItem>);
        expect(container).toMatchSnapshot();
    })
})