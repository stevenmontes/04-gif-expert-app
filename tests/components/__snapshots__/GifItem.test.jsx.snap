// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests of GifItem should match with the snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="title"
      src="url"
    />
    <p>
      title
    </p>
  </div>
</div>
`;
